src/features/marketplace/process/clients/marketplace-client-creator.ts

```diff
@@ -0,0 +1,15 @@
+import ContractEvent from '../enums/contract-event.enum';
+import IClient from '../interfaces/client.interface';
+import MarketplaceState from '../types/marketplace-state';
+import ListItemClient from './list-item-client';
+import BuyItemClient from './buy-item-client';
+
+export default class MarketplaceCreator {
+  static createClient(state: MarketplaceState): IClient {
+    if (state.mintEventName === ContractEvent.LogCreateItem) return new ListItemClient(state);
```

Comment:
Change the name of the variable. This is not a mint anymore. Something like listingEvent, or whatever the process is called.